android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,196 @@
+                    val notification = notifyForegroundService(applicationContext)
+                    if (breezSDK == null) {
+                        startForeground(NOTIFICATION_ID_FOREGROUND_SERVICE, notification)
+                        breezSDK = connectSDK(applicationContext, SDKListener())
+                    }
+                    pollerScope.launch(
+                        Dispatchers.IO +
+                                CoroutineExceptionHandler { _, e ->
+                                    Logger.tag(TAG).error { "Error when polling payment: $e" }
+                                    notifyPaymentFailed(applicationContext)
+                                }
+                    ) {
+                        pollPaymentHash(paymentHash, clickAction)
+                    }
+                }
+            }
+        }
+        return START_NOT_STICKY
+    }
+
+    private suspend fun pollPaymentHash(
+        paymentHash: String,
+        clickAction: String?,
+    ) {
+        if (!paymentHashPollerList.contains(paymentHash)) {
+            paymentHashPollerList.add(paymentHash)
+        } else {
+            throw Exception("Payment w/ hash: $paymentHash is already being polled.")
+        }
+        Logger.tag(TAG).debug { "Polling for payment w/ hash: $paymentHash" }
+        // Poll for 1 minute
+        for (i in 1..60) {
+            val payment = breezSDK!!.paymentByHash(paymentHash)
+            if (payment?.status == PaymentStatus.COMPLETE) {
+                paymentHashPollerList.remove(paymentHash)
+                val pd = payment.details as PaymentDetails.Ln
+                val lnPD = pd.data
+                handleReceivedPayment(
+                    lnPD.bolt11,
+                    lnPD.paymentHash,
+                    payment.amountMsat,
+                    clickAction
+                )
+                return
+            } else {
+                Logger.tag(TAG).info { "Payment w/ paymentHash: $paymentHash not received yet." }
+                withContext(Dispatchers.IO) { Thread.sleep(1_000) }
+            }
+        }
+
+        // If we reach here then we didn't receive for more than 1 minute
+        paymentHashPollerList.remove(paymentHash)
+        throw Exception("Payment not found before timeout")
+    }
+
+    private fun handleReceivedPayment(
+        bolt11: String,
+        paymentHash: String,
+        amountMsat: ULong?,
+        clickAction: String? = null,
+    ) {
+        Logger.tag(TAG).info {
+            "Received payment. Bolt11:${bolt11}\nPayment Hash:${paymentHash}"
+        }
+        val amountSat = (amountMsat ?: ULong.MIN_VALUE) / 1000u
+        notifyPaymentReceived(applicationContext, clickAction = clickAction, amountSat = amountSat)
+
+        // Push back shutdown by SHUTDOWN_DELAY_MS in case we'll receive more payments
+        shutdownHandler.removeCallbacksAndMessages(null)
+        shutdownHandler.postDelayed(shutdownRunnable, SHUTDOWN_DELAY_MS)
```

Comment:
Seems like we will not get any "Failed payment" notification but I think it is good behavior that also simplify the flow so let's keep it this way.

Reply:
Saw these comments a bit late after making changes... 

We're no longer polling for payment hash & displaying any failed payment notifications anymore.

We're delaying the shutdown by 2 minutes every time we receive a payment but this was taking poller into consideration, which ran for a 1 minute, so we should reduce shutdown delay back to 1 minute.